### PR TITLE
Bidirectional RNNs Debug

### PR DESCRIPTION
**Pull request by jedi00**
#453 , @iskandr submitted  a 'Bidirectional‘ RNN class. This submission resolves the bug in @iskandr 's code.

### CODE REVIEW

**Comment by iskandr** [comment]
@jedi00 thanks! Can you explain what the bug and fix are?

**Comment by jedi00** [comment]
@iskandr : I've described the bug under your #453 request.
And, there's still a problem: the bidirectional class is not compatible with dropout class. I'm working on this but still no clue. Besides, I've not tested other layer's compatibility.

File: keras/layers/recurrent.py [hunk 1]
```diff
@@ -33,6 +33,35 @@ def get_padded_shuffled_mask(self, train, X, pad=0):
             mask = T.concatenate([padding, mask], axis=0)
         return mask.astype('int8')
 
+class Transparent(Recurrent):
```
Comment (iskandr) [inline keras/layers/recurrent.py]: Is there any way to connect the input to RNNs without using a dummy layer?

**Comment by fchollet** [comment]
Ok, so we have 2 outstanding PRs for bidirectional RNNs. Should I only look at this one, or do you want to merge both into one PR for proper attribution?

**Comment by iskandr** [comment]
@jedi00 Did you figure out what goes wrong with Dropout? Do you want me to combine this fix with my PR or should I close mine?

**Comment by jedi00** [comment]
@iskandr : Unfortunately no. I'll close this pull request.
